python/ray/serve/handle.py

```diff
@@ -259,14 +278,20 @@ def _fetch_future_result_sync(
         """
 
         if self._replica_result is None:
-            try:
-                self._replica_result = self._replica_result_future.result(
-                    timeout=_timeout_s
+            if self._is_router_running_in_separate_loop:
+                try:
```

Comment:
nit: can short circuit, less nested code
```suggestion
            if not self._is_router_running_in_separate_loop:
                raise RuntimeError(
                    "Sync methods should not be called from within an `asyncio` event "
                    "loop. Use `await response` instead."
                )
                
            try:
```